valohai_yaml/commands.py

```diff
@@ -44,22 +48,27 @@ def _replace_interpolation(parameter_map: ParameterMap, match: 'Match[str]') ->
 def build_command(
     command: Union[str, List[str]],
     parameter_map: ParameterMap,
+    special_interpolations: Optional[Dict[str, str]] = None,
 ) -> List[str]:
     """
     Build command line(s) using the given parameter map.
 
-    Even if the passed a single `command`, this function will return a list
+    Even if passed a single `command`, this function will return a list
     of shell commands.  It is the caller's responsibility to concatenate them,
     likely using the semicolon or double ampersands.
 
     :param command: The command to interpolate params into.
     :param parameter_map: A ParameterMap object containing parameter knowledge.
+    :param special_interpolations: a str-str dict containing special interpolations
 
     :return: list of commands
     """
     if isinstance(parameter_map, list):
         raise TypeError("Passing in lists as ParameterMaps is no longer supported.")
 
+    special: Dict[str, str]
+    special = {} if special_interpolations is None else special_interpolations
```

Comment:
This makes me a bit sad. Isn't mypy placated by 

```suggestion
    if special_interpolations is None:
        special_interpolations = {}
```
?

Reply:
it is not, hence the spell casting